### DISCUSSION TITLE
Filtering stops working on all clients

### DISCUSSION THREAD

**Discussion by moofawsaw** [AdguardTeam/AdGuardHome#2820] [Q&A]
After installing on a Mac M1 the filter was working fine on all devices however after a few hours traffic stopped being filtered through Adguard Home. 

Is there a step in the configuration that I am missing? I have set my DNS to 192.168.0.11 in my router settings and it was working fine for a few hours until nothing was being routed through Adguard. I rebooted the router remotely and the filters were working for a few hours and then stopped again. 

Here are my configuration settings:

127.0.0.1
::1
192.168.0.11
2607:fea8:44df:420:18bd:d3d6:a69a:95ac
2607:fea8:44df:420:f8ff:7f10:d5f5:31ce
2607:fea8:44df:420::820

Checking the JSON file I notice that the Upstream DNS seems to be switching between the default and the custom DNS I have set. Note that https://dns10.quad9.net:443/dns-query is not listed as an upstream DNS server. 

{"IP":"192.168.0.14","T":"2021-02-01T09:46:31.929558-05:00","QH":"1p.ca.telus.rcs.telephony.goog","QT":"AAAA","QC":"IN","CP":"","Answer":"/cmBgAABAAAAAQAAAjFwAmNhBXRlbHVzA3Jjcwl0ZWxlcGhvbnkEZ29vZwAAHAABwBwABgABAAAAPAAwA25zMQZnb29nbGUDY29tAAlkbnMtYWRtaW7AQBUlkz8AAAOEAAADhAAABwgAAAA8","Result":{},"Elapsed":41237791,"Upstream":"https://dns10.quad9.net:443/dns-query"}

{"IP":"192.168.0.14","T":"2021-02-01T09:49:30.569926-05:00","QH":"sa.api.intl.miui.com","QT":"AAAA","QC":"IN","CP":"","Answer":"6ZWBgwABAAAAAQAAAnNhA2FwaQRpbnRsBG1pdWkDY29tAAAcAAHADAAGAAEAAA4QAD4NY2xlYW5icm93c2luZwNycHoDbm9jA29yZwAMYWNjZXNzcG9saWN5wEAAAAABAAAcIAAAA4QAEnUAAAFRgA==","Result":{},"Elapsed":64418000,"Upstream":"https://doh.cleanbrowsing.org:443/doh/custom-filter/7dcb:1b94:96eb:4fe4/"}

Note that when I reboot the router, the Mac IP processed the following and then never received any traffic after that:

9-courier.push.apple.com (1 times)
Type: A, Plain DNS

11-courier.push.apple.com (6 times)
Type: HTTPS, Plain DNS

gspe1-ssl.ls.apple.com (2 times)
Type: A, Plain DNS

**Comment by ameshkov** [comment]
Could it be that your Mac IP address changes and it stops receiving queries?

**Comment by moofawsaw** [comment]
If my Mac IP address is changing I am not actively doing it. I ran the terminal command `ipconfig getifaddr en0` and found that my Mac IP matches what is displayed on the configuration in the setup guide:

192.168.0.11

**Comment by moofawsaw** [comment]
After a second reboot here is every processes from the Query Log that ran once, all at the same time:

ipv4only.arpa
Type: A, Plain DNS

example.org
Type: AAAA, Plain DNS

example.org
Type: A, Plain DNS

detectportal.firefox.com
Type: A, Plain DNS

detectportal.firefox.com
Type: AAAA, Plain DNS

lb._dns-sd._udp.ht.home
Type: PTR, Plain DNS

ipv4only.arpa
Type: AAAA, Plain DNS

lb._dns-sd._udp.0.0.168.192.in-addr.arpa
Type: PTR, Plain DNS

11.0.168.192.in-addr.arpa
Type: PTR, Plain DNS

**Comment by ameshkov** [comment]
Tbh, I have no clue what's happening. You may need to check your router logs to figure out why it behaves like this.

**Comment by moofawsaw** [comment]
I tried installing on two different Mac M1's and setting my DNS on my router to match the IP of each device and the same issue happens on both devices. I have rebooted the router and uninstalled / reinstalled Adguard Home and the issues remains. Adguard filters traffic coming via the Mac's for a little bit then stops. Traffic is still being router from mobile devices but not from these Mac M1's.

Note that when I have Adguard Home installed and configured on one Mac and try testing on other, traffic still doesn't come from the non-installed Mac either. This leads me to believe that it is a Mac issue since traffic is still being routed from Mobile devices.

**Comment by ameshkov** [comment]
Have you tried setting it up on a non-M1 device? Tbh, I am still not sure the problem is in M1 and not in the router or smth else.

**Comment by moofawsaw** [comment]
Before I bought these M1's I was up and running on my 2015 Intel Mac running the same OS. I didn't change anything in my router settings during this time.

**Comment by ameshkov** [comment]
I guess to properly troubleshoot it you should try running `nslookup example.org AG_IP_ADDRESS` command when it happens to see if it still listens to that address.

**Comment by moofawsaw** [comment]
Here is the result of the command: 

kyleunderhill@MacBook-Air ~ % nslookup example.org AG_IP_ADDRESS
nslookup: couldn't get address for 'AG_IP_ADDRESS': not found

**Comment by ameshkov** [comment]
By AG_IP_ADDRESS I meant the IP address of AdGuard Home

**Comment by moofawsaw** [comment]
kyleunderhill@MacBook-Air ~ % nslookup example.org 192.168.0.11
Server:		192.168.0.11
Address:	192.168.0.11#53

Non-authoritative answer:
Name:	example.org
Address: 93.184.216.34

**Comment by ameshkov** [comment]
Well, so it works after all. This proves that the problem is not in AGH but somewhere else.

**Comment by moofawsaw** [comment]
I factory reset the Macbook that wasn't `192.168.0.11` and traffic still wasn't being filtered from that device. AGH is set up and working on my android no problem, but the Mac's aren't being routed through AGH.

It wouldn't be an issue with my router since AGHit is working on Android. It must be a Mac M1 specific problem because it was working fine on the Intel Mac I had before.

**Comment by moofawsaw** [comment]
UPDATE*

I purchased a Raspberry Pi and setup Adguard Home to run through the device. I then uninstalled AGH on the Mac M1 and rebooted the router. It started blocking traffic coming from the M1 for a few hours even over night until it just stopped. No traffic coming from the M1 is being routed through the Raspberry Pi IP set up with AGH 192.168.0.16.

Note however that traffic is being blocked coming from Android devices. 

This leads me to believe that the issue is specifically with the M1. Could something be happening when I access AGH from the Mac M1 using the address for the Raspberry Pi? http://192.168.0.16/

The only thing I can think is that logging into the AGH interface from the M1 is somehow throwing everything off.

**Comment by buarle** [comment]
Had a similar issue with AGH on raspi4 Running as a docker container.

I solved it via downgrading from v0.105.x to v0.104.3...

Best

**Comment by moofawsaw** [comment]
Now the Raspberry Pi isn't blocking traffic from any device according to the Query Log. Something seriously wrong here.

**Comment by ameshkov** [comment]
> No traffic coming from the M1 is being routed through the Raspberry Pi IP set up with AGH 192.168.0.16

This kinda bothers me. Are there simply no queries from M1 in AGH's Query Log at that time?

> Now the Raspberry Pi isn't blocking traffic from any device according to the Query Log.

Or do you see queries and none of them are getting blocked?

**Comment by ghost** [comment]
I have the same problem on a raspberry pi. Nothing is blocked for days after updating to 1.105.x. getting queries in from different clients but not a single one is blocked.
But I do get malware/Phishing blocked.

I additionally open a Filterlist and opened blocked domains directly but they don't getting blocked.

Checked e.g. abc-ads.com which is def in one of the filterlists.
![Screenshot_20210313-180825](https://user-images.githubusercontent.com/22637669/111038104-84b55780-8427-11eb-81e3-3b1bb8fefb83.png)
![Screenshot_20210313-181025](https://user-images.githubusercontent.com/22637669/111038106-8848de80-8427-11eb-9f43-7b4ef3e6ed9e.png)
![Screenshot_20210313-181155](https://user-images.githubusercontent.com/22637669/111038130-9dbe0880-8427-11eb-8b64-d059fa27daee.png)

**Comment by ameshkov** [comment]
@ConfiDev

First of all, try disabling all filter lists and keep just the default one to see if it changes anything.
If it does, try enabling them back one by one to find out which one is responsible.

**Comment by ghost** [comment]
@ameshkov 
Thanks for your reply.
I did what you asked for and got the filtering running again.
Deactivated everything and got queries blocked again. Interestingly enough I enabled all the filter lists and it seemed to work fine.
Later i removed a list with a false positive but the false positive was still there but in an "unknown list" and it stayed that way. Then I disabled the last list I added with over 2 million entires and the false positive in the unknown list was gone.
So either its this one list or the amount of entries. Even before this list I had about 1.5 - 2 million entries.

I remembered that I ones tried to enable a list with over 10 million entries which killed my rasp pi tho.

**Comment by moofawsaw** [comment]
I tried configuring Pi-Hole to test whether it was a router vs. platform / M1 issue and the same thing happened through pi-hole as was reported here. It worked for a day or so and then stopped filtering traffic in the exact same way AGH did. 

Unfortunately for me the issue is with my router and I have no idea how to troubleshoot it but I guess you can mark this ticket as closed since it doesn't seem to be an AGH issue.

**Comment by ameshkov** [comment]
Moved to Q&A since this is not a bug of AGH.

Regarding troubleshooting, maybe you should try Googling your router name + problem description, may be a known issue.
